src/main/java/org/myrobotlab/service/abstracts/AbstractPinEncoder.java

```diff
@@ -78,37 +78,32 @@ public void setZeroPoint() {
     // pass the set zero point command to the controller
     controller.setZeroPoint(this);
   }
-
-  @Override
-  public void setController(EncoderController controller) {
-    this.controller = controller;
-  }
-
-  @Override
+  
   public void setPin(String pin) {
     this.pin = pin;
   }
 
-  @Override
   public void setPin(Integer address) {
     this.pin = String.format("%d", address);
   }
 
   @Override
   public void disable() {
-    // TODO Auto-generated method stub
-    
+    enabled = false;
   }
 
   @Override
   public void enable() {
-    // TODO Auto-generated method stub
-    
+    enabled = true;

```

Comment:
this doesn't actually tell the underlying encoder to be enabled.  as mentioned in other comments.  this should be abstract

Reply:
same as above - my sloppy first interface definition, meets your first implementation(s)